pkg/service/service.go

```diff
@@ -319,6 +327,43 @@ func (s *Service) InitMaps(ipv6, ipv4, sockMaps, restore bool) error {
 	return nil
 }
 
+func (s *Service) upsertEnvoyEndpoints(svc *svcInfo) error {
+	name := svc.svcNamespace + "/" + svc.svcName
+	lbEndpoints := []*envoy_config_endpoint.LbEndpoint{}
+	for _, be := range svc.backends {
+		lbEndpoints = append(lbEndpoints, &envoy_config_endpoint.LbEndpoint{
+			HostIdentifier: &envoy_config_endpoint.LbEndpoint_Endpoint{
+				Endpoint: &envoy_config_endpoint.Endpoint{
+					Address: &envoy_config_core.Address{
+						Address: &envoy_config_core.Address_SocketAddress{
+							SocketAddress: &envoy_config_core.SocketAddress{
+								Address: be.L3n4Addr.IP.String(),
+								PortSpecifier: &envoy_config_core.SocketAddress_PortValue{
+									PortValue: uint32(be.L3n4Addr.L4Addr.Port),
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+	}
+	endpoint := &envoy_config_endpoint.ClusterLoadAssignment{
+		ClusterName: name,
+		Endpoints: []*envoy_config_endpoint.LocalityLbEndpoints{
+			{
+				LbEndpoints: lbEndpoints,
+			},
+		},
+	}
+	var resources envoy.Resources
+	resources.Endpoints = append(resources.Endpoints, endpoint)
+	if s.monitorNotify != nil {
+		return s.monitorNotify.UpsertEnvoyResources(context.TODO(), resources, false)
```

Comment:
Done by adding a new `envoyCache` interface with only the functions that were actually used. Also rearranged daemon bootstrap a bit so that `d.l7Proxy` is initialized (if enabled) before the `NewService()` call.